weaviate_cli/managers/config_manager.py

```diff
@@ -127,12 +127,12 @@ def get_client(self) -> weaviate.WeaviateClient:
             host_parsed = urlparse(self.config["host"])
             grpc_parsed = urlparse(self.config["grpc_host"])
             return weaviate.connect_to_custom(
-                http_host=host_parsed.hostname,
-                grpc_host=grpc_parsed.hostname,
+                http_host=host_parsed.hostname or self.config["host"],
+                grpc_host=grpc_parsed.hostname or self.config["grpc_host"],
                 grpc_secure=grpc_parsed.scheme == "https",
                 http_secure=host_parsed.scheme == "https",
-                http_port=self.__get_port(host_parsed),
```

Comment:
Let's do one thing, maybe we can define the following logic:
In __get_port, 
```
    def __get_port(self, url_parsed):
        if url_parsed.port:
            return url_parsed.port
        if url_parsed.scheme == "https":
            return 443
        else:
            return 80
```
if self.config["http_port"] or self.config["grpc_port"] is not empty, we return the value from self.config, otherwise we try to infer it from host_parsed and grpc_parsed, this way we don't break backwards compatibility.

http_port=self.config["http_port"] if self.config["http_port"] != "" else self.__get_port(host_parsed)
grpc_port=self.config["grpc_por"] if self.config["grpc_port"] != "" else self.__get_port(grpc_parsed)